apps/agent/README.md

```diff
@@ -0,0 +1,254 @@
+
+#### 3.1.1. Execute 
+ - **Name:** `Execute`
+ - **Args:** 
+   - **Sender:** Address of the original sender
+   - **Target:** Address of the target contract
+   - **ETH value:** ETH amount sent in Wei
+   - **Data:** Transaction data
+
+#### 3.1.2. Safe execute
+ - **Name:** `SafeExecute`
+ - **Args:** 
+   - **Sender:** Address of the sender
+   - **Target:** Address of the target contract
+   - **Data:** Transaction data
+
+#### 3.1.3. Add protected token 
+ - **Name:** `AddProtectedToken`
+ - **Args:** 
+   - **Token:** Token address
+
+#### 3.1.4. Remove protected token 
+ - **Name:** `RemoveProtectedToken`
+ - **Args:** 
+   - **Token:** Token address
+
+#### 3.1.5. Presign hash 
+ - **Name:** `PresignHash`
+ - **Args:** 
+   - **Sender:** Address of the sender
+   - **Hash:** Presigned hash
+
+#### 3.1.6. Set designated signer
+ - **Name:** `SetDesignatedSigner`
+ - **Args:** 
+   - **Sender:** Address of the sender
+   - **Old signer:** Address of the previous signer
+   - **New signer:** Address of the new signer
+
+### 3.2. Getters
+
+#### 3.2.1. Is presigned
+ - **Inputs:** 
+   - **Hash:** Hash
+ - **Pre-flight checks:** None
+ - **Outputs:**
+   - **Is presigned**: Whether the hash is presigned or not.
+
+#### 3.2.2. Designated signer
+ - **Inputs:** None
+ - **Pre-flight checks:** None
+ - **Outputs:**
+   - **Designated signer**: Address of the designated signer
+
+#### 3.2.3. Protected token
+ - **Inputs:** 
+   - **Index:** Token index
+ - **Pre-flight checks:** None
+ - **Outputs:**
+   - **Protected tokens**: Address of the protected token
+
+#### 3.2.4. Is forwarder
+ - **Inputs:** None
+ - **Pre-flight checks:** None
+ - **Outputs:**
+   - **Is forwarder**: Whether `Agent` is a `Forwarder`. Always `true`.
+
+#### 3.2.5. Can forward
+ - **Inputs:** 
+   - **Sender:** Address of the account intending to forward an action
+   - **EVM script**: Script to forward
+ - **Pre-flight checks:** None
+ - **Outputs:**
+   - **Can forward**: Whether the action can be forwarded or not
+
+#### 3.2.6. Protected tokens length
+ - **Inputs:** None
+ - **Pre-flight checks:** 
+   - Ensure the contract is initialized
+ - **Outputs:**
+   - **Protected tokens length**: Number of protected tokens
+
+#### 3.2.7. Supports interface
+ - **Inputs:** 
+   - **Interface ID:** Interface bytes to check
+ - **Pre-flight checks:** None
+ - **Outputs:**
+   - **Supports interface**: Whether the contract supports the interface
+
+#### 3.2.8. Is valid signature
+ - **Inputs:** 
+   - **Hash:** Data signed on the behalf of address(this)
+   - **Signature:** Signature byte array associated with the data
+ - **Pre-flight checks:** None
+ - **Outputs:**
+   - **Is valid signature bytes**: The `ERC-1271` magic value if the signature is valid
+
+
+## 4. Architecture
+
+### 4.1. Inheritance and implementation
+
+- [Vault](../vault/contracts/Vault.sol)
+- [Forwarder](https://github.com/aragon/aragonOS/blob/master/contracts/common/IForwarder.sol)
```

Comment:
Not a strong opinion; maybe linking to the hack.aragon.org documentation on forwarders would be better?